python/cudf/cudf/core/frame.py

```diff
@@ -2309,7 +2309,7 @@ def drop_duplicates(
         return result
 
     def replace(self, to_replace: Any, replacement: Any) -> Frame:
-        copy_data = self._data.copy()
+        copy_data = self._data.copy(deep=True)
```

Comment:
We actually don't want to deep copy here, right? Given we go through and run a `find_and_replace` on each column that returns by deep copy?

Reply:
So the reason I had this changed to deep copy was for when there are these kinds of inputs:
```python
df.update({'a':2}, {'c':10})
```
where the `to_replace` and `replacement` keys are not symmetric all the maps returned by `_get_replacement_values_for_columns` will be empty and thus we could end up returning shallow copied columns without even calling `find_and_replace`. 

So to avert this situation, In try/catch block I'm doing a deep copy of the column whereas the initial column accessor is changed to a shallow copy now.

Reply:
Perfect fix, thanks!